esmvaltool/diag_scripts/weighting/climwip.py

```diff
@@ -371,12 +372,71 @@ def compute_overall_mean(dataset: 'xr.Dataset',
     return overall_mean
 
 
+def combine_dimension(dataset: 'xr.DataArray', dimn: str) -> 'xr.DataArray':
+    """Combine ensemble members of the same model along the given dimension."""
+    model_ensemble = dataset[dimn].values
+    models = [me.split('_')[0] for me in model_ensemble]
+    dataset['model'] = xr.DataArray(models, dims=dimn)
+    dataset = dataset.groupby('model').mean(keep_attrs=True).rename(
+        {'model': dimn})
+
+    # new model names
+    groups = {
```

Comment:
This is a lot of hassle to keep track of all ensemble members that are part of each of the models. And then you have to infer it again in `split_ensemble_members`. 
I wonder if it would be easier to just return the groups dictionary and pass it explicitly to the splitting function.

Reply:
What do you think of something like:

```python
groups = defaultdict(list)
for name in dataset[dimn].values:
    model, ensemble = name.split('_')[:2]
    groups[model].append(ensemble)

groups

> defaultdict(list,
>            {'ACCESS1-0': ['r1i1p1'],
>             'ACCESS1-3': ['r1i1p1'],
>             'BNU-ESM': ['r1i1p1'],
>             'CCSM4': ['r1i1p1', 'r2i1p1', 'r3i1p1', 'r4i1p1']})
```

Reply:
You could also combine that with 
```python
models = []
for ....
    models.append[model]
```
which you can then use to group the models with. Effectively merging the implicit loop from line 378.

Reply:
so essentially, reading the models, ensembles from the recipe and use this information? makes sense to me, at least as long as there are no plans to allow anything like "use all available model runs" (which is what we do in ClimWIP and thats why this is implemented as it is now I guess).

Reply:
I like the idea with the dictionary, simplifies the code and makes it more understandable! Thanks for this!!